### ISSUE TITLE
How to provide an image pull secret when using "--image-repository" with an internal registry that does not allow anonymous pull requests?

### ISSUE THREAD

**Issue by jgoeres** [kubernetes/minikube#10019]
After a bit of a hiatus, I have been revisiting the whole topic of running minikube from purely local sources (i.e., any images should come from our own private registry). I revisted the issue https://github.com/kubernetes/minikube/issues/8298 that I created in June. The first problem I am running into is that our internal registry does not support anonymous pull access.
In the above mentioned issue, @tstromberg mentioned two sources on how to provide a pull secret - but all of the already require a running cluster, as they both boil down to enabling the registry-creds plugin:

    minikube addons configure registry-creds

But at that point all the infrastructure images would already have to be there, no?
Further, this add-on only allows me to specify creds for ECR, GCR, Dockerhub or ACR - what about my own registry?

**Comment by medyagh** [comment]
good idea, I would like to fully support this and possibly add integraiton test for it, there is a WIP PR for making addons possible with custom image.


https://github.com/kubernetes/minikube/pull/10111

**Comment by medyagh** [comment]
> Further, this add-on only allows me to specify creds for ECR, GCR, Dockerhub or ACR - what about my own registry?

that is a good point, do u know if other simmilar tools using a specific tool that minikube could take advantage of ?
in the past I have seen people deploy a kubertnes secret to their Pod, that will let the pod be able to pull from a specific registery, have u already that ?

**Comment by jgoeres** [comment]
Forour own product deployment (in an already running cluster) we simply provide allow to provide the registry creds, which go into a secret passed to all pods as an image pull secret. But from my understanding, since we are in the process of bootstrapping the cluster, this probably has to be an "out of band" mechanism.

**Comment by sharifelgamal** [comment]
Freezing this as this is definitely a feature we would accept a PR for.
